Document `computeEditInfo` option (#8486)
stylelint/stylelint: docs/user-guide/options.md

- a scoped disable comment, e.g. `/* stylelint-disable color-named */`, any problems reported by the scoped rule will not be automatically fixed anywhere in the source
- an unscoped disable comment, i.e. `/* stylelint-disable */`, the entirety of the source will not be automatically fixed for that rule

## `computeEditInfo`

CLI flag: `--compute-edit-info, --cei`

Compute edit information for autofixable rules.

When enabled, warnings from autofixable rules will include a `fix` property containing:

- `range` - the start and end indices of the text to replace
- `text` - the replacement text

The edit information will not be computed when:

> - the `fix` option is enabled

Reviewer: nitpick

Maybe link the option?

```suggestion
- the [`fix` option](#fix) is enabled
```

- a rule's fix is disabled:
  - in the configuration object, e.g. `"rule-name": [true, { disableFix: true }]`

>   - using disable comments, e.g. `/* stylelint-disable rule-name */`

Reviewer: ```suggestion
  - using configuration comments, e.g. `/* stylelint-disable rule-name */`
```

cf https://stylelint.io/user-guide/configure/#configurationcomment

>   - using configuration comments, e.g. `/* stylelint-disable rule-name */`

Reviewer: ```suggestion
  - using configuration comments, e.g. `/* stylelint-disable rule-name */` (if [`ignoreDisables`](#ignoredisables) is `false`)
```

Do we need a test for that?
i.e. `ignoreDisables: true` + configuration comment

Reviewer: @jeddy3 is this too obvious?

- another edit has already been computed for the same code region

> 
> See [Warning with EditInfo](errors.md#warning-with-editinfo) for examples.

Reviewer 2: ```suggestion
```

> NODE_OPTIONS='--disable-warning=stylelint:005' npx stylelint "**/*.css"
> ```

## `computeEditInfo`

CLI flag: `--compute-edit-info, --cei`

> Record edit information for autofixable rules without applying the fixes.
> 
> When enabled, warnings from autofixable rules will include a `fix` property containing:
> 
> - `range` - the start and end indices of the text to replace
> - `text` - the replacement text
> 
> The edit information will not be recorded in these cases:
> 
> - When `fix` option is enabled (`fix` takes precedence over `computeEditInfo`)
> - When the rule's fix is disabled:
>   - via configuration (`"rule-name": [true, { disableFix: true }]`)
>   - via disable comments (`/* stylelint-disable rule-name */`)
> - When another edit is already recorded for the same code region

Reviewer 2: ```suggestion
Compute edit information for autofixable rules.

When enabled, warnings from autofixable rules will include a `fix` property containing:

- `range` - the start and end indices of the text to replace
- `text` - the replacement text

The edit information will not be computed when:

- the `fix` option is enabled
- a rule's fix is disabled:
  - in the configuration object, e.g. `"rule-name": [true, { disableFix: true }]`
  - using disable comments, e.g. `/* stylelint-disable rule-name */`
- another edit has already been computed for the same code region
```

Some minor tweaks to align with our content style.

> See [Warning with EditInfo](errors.md#warning-with-editinfo) for examples.

Author: Will be added in this PR: https://github.com/stylelint/stylelint/pull/8485
